### ISSUE TITLE
fill-config turns "source = ..." into "factory = ..."

### ISSUE THREAD

**Issue by Mwni** [explosion/spaCy#7055]
Let's say we have this base config:
```
[components]

[components.tok2vec]
source = "en_core_web_trf"

[components.transformer]
source = "en_core_web_trf"

[components.parser]
source = "en_core_web_trf"

[components.tagger]
source = "en_core_web_trf"

[components.ner]
factory = "ner"
```
Which effectively is supposed to state, that I want a model that reuses all components of en_core_web_trf, except for the NER.
Running it through `py -m spacy init fill-config base_config.cfg config.cfg` however yields a config that defines each component as factory, basically ignoring what was stated before:
```
[components]

[components.ner]
factory = "ner"

[components.ner.model]
@architectures = "spacy.TransitionBasedParser.v2"
state_type = "ner"
...

[components.parser]
factory = "parser"
learn_tokens = false
...

[components.parser.model]
@architectures = "spacy.TransitionBasedParser.v2"
state_type = "parser"
...

[components.tok2vec]
factory = "tok2vec"

ETC...
```
What's going on? Did I misinterpret the documentation?

Windows, Python 3.8, spacy 3.0.0

**Comment by svlandeg** [comment]
Thanks for the detailed bug report! This was a bug indeed, and is fixed by PR #7058. The fix is available in spaCy 3.0.3 which was released a few hours ago :-)

**Comment by Mwni** [comment]
No, thank YOU for the record speed fix of this :)
